reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md

```diff
@@ -0,0 +1,117 @@
+allowed to be breaking.
+
+Experimental features are disabled by default and need to be explicitly enabled
+by the user or administrator of the system.
+
+Enabled experimental features are listed in the `powershell.config.json` file
+in `$PSHOME` for all users or the user-specific configuration file for a
+specific user.
+
+> [!NOTE]
+> Experimental features enabled in the user configuration file take precedence
+> over experimental features listed in the system configuration file.
+
+## Declaring Experimental Features in Modules written in C#
+
+Module authors who want to use the Experimental Feature flags can declare
+a cmdlet as experimental by using the `Experimental` attribute:
+
+```csharp
+[Experimental("PSWebCmdletV2", ExperimentAction.Show)]
+[Cmdlet(Verbs.Invoke, "WebRequest")]
+public class InvokeWebRequestCommandV2 : WebCmdletBaseV2 { ... }
+```
+
+## Declaring Experimental Features in Modules written in PowerShell
+
+Module written in PowerShell can also use the `Experimental` attribute to
+declare experimental cmdlets:
+
+```powershell
+function Enable-SSHRemoting {
+    [Experimental("PSSSHRemoting", "Show")]
+    [CmdletBinding()]
+    param()
+    ...
+}
+```
+
+### Mutually Exclusive Experimental Features
+
+You can have an experimental cmdlet that overrides an existing cmdlet. The two
+versions can't coexist side by side. The `ExperimentAction.Hide` setting
+allows only one of the two cmdlets to be enabled at one time.
+
+In this example, we create a new experimental `Invoke-WebRequest` cmdlet.
+`InvokeWebRequestCommand` contains the non-experimental implementation.
+`InvokeWebRequestCommandV2` contains the experimental version of the cmdlet.
+
+```csharp
+[Experimental("PSWebCmdletV2", ExperimentAction.Show)]
+[Cmdlet(Verbs.Invoke, "WebRequest")]
+public class InvokeWebRequestCommandV2 : WebCmdletBaseV2 { ... }
+
+[Experimental("PSWebCmdletV2", ExperimentAction.Hide)]
+[Cmdlet(Verbs.Invoke, "WebRequest")]
+public class InvokeWebRequestCommand : WebCmdletBase { ... }
+```
+
+When the `PSWebCmdletV2` experimental feature is enabled, the existing
+`InvokeWebRequestCommand` implementation is hidden and the
+`InvokeWebRequestCommandV2` provides the implementation of
+`Invoke-WebRequest`.
+
+This allows users to try out the new cmdlet and provide feedback then revert
+to the non-experimental version when needed.
+
+### Experimental Parameters in Cmdlets
+
+The `Experimental` attribute can also be applied to individual parameters.
+This allows you to create an experimental set of parameters for an existing
+cmdlet rather than an entirely new cmdlet.
+
+```csharp
+[Experimental("PSNewAddTypeCompilation", ExperimentAction.Show)]
+[Parameter(ParameterSet = "NewCompilation")]
+public CompilationParameters CompileParameters { ... }
+
+[Experimental("PSNewAddTypeCompilation", ExperimentAction.Hide)]
+[Parameter()]
+public CodeDom CodeDom { ... }
+```
+
+```powershell
+param(
+    [Experimental("PSNewFeature", "Show")]
```

Comment:
Are these two examples related? If so, the experiment names don't match. If not the we need to make it clearer that these are separate examples.